tests/acceptance/test_update_control.py

```diff
@@ -615,3 +616,96 @@ def wait_for_state(state_to_wait_for):
+    def test_many_state_transitions_with_update_control(
+        self,
+        setup_board,
+        connection,
+        second_connection,
+        setup_mock_server,
+        bitbake_variables,
+        bitbake_path,
+    ):
+        """Test whether we can make many state transitions with update control without
+        triggering the "too many state transitions" error."""
+
+        try:
+            start_and_ready_mender_client(connection, second_connection)
+
+            ucm = (
+                """{
+    "ID": "%s",
+    "states": {
+        "ArtifactInstall_Enter": {
+            "action": "pause"
+        }
+    }
+}"""
+                % MUID
+            )
+
+            with tempfile.TemporaryDirectory() as tmpdir:
+                script = os.path.join(tmpdir, "map-insert.sh")
+                with open(script, "w") as fd:
+                    fd.write(
+                        """#!/bin/sh
+while sleep 0.2; do
+    dbus-send --system --dest=io.mender.UpdateManager --print-reply /io/mender/UpdateManager io.mender.Update1.SetUpdateControlMap string:'%s'
+done
+"""
+                        % ucm
+                    )
+                put_no_sftp(script, connection, remote="/data/map-insert.sh")
+
+            # Constantly reinsert map over and over in the background, to force
+            # state transitions.
+            connection.run("systemd-run sh /data/map-insert.sh")
+
+            now = time.time()
+
+            make_and_deploy_artifact(
+                connection, bitbake_variables["MENDER_DEVICE_TYPE"]
+            )
+
+            timeout = now + 300
+            # Wait until we have received 100 state transitions, which is way
+            # more than would cause a failure.
+            while time.time() < timeout:
+                output = connection.run(
+                    "journalctl -u mender-client -S '%s' | grep 'State transition: mender-update-control '"
+                    % time.strftime("%Y-%m-%d %H:%M:%S UTC", time.gmtime(now)),
+                    warn=True,
+                ).stdout
+                if len(output.split("\n")) >= 100:
+                    break
+            else:
+                pytest.fail(
+                    "Timed out without reaching the required number of state transitions."
+                )
+
+            # Just double check that we are indeed paused, as we expect.
+            log = wait_for_state(connection, "Download")
+            assert "Cleanup" not in log
+
+            # Force it to continue.
+            set_update_control_map(
+                connection,
+                {
+                    "ID": MUID2,
+                    "priority": 10,
+                    "states": {"ArtifactInstall_Enter": {"action": "force_continue"}},
+                },
+            )
+
+            # Rest of deployment should finish successfully.
+            log = wait_for_state(connection, "Cleanup")
+            assert "ArtifactFailure" not in log
+
```

Comment:
I suggest adding there the same `except` we have for the other tests:
```
        except:
            connection.run("journalctl -u mender-client | cat")
            raise
```

Reply:
Fixed!